nixd/lib/CommandLine/Configuration.cpp

```diff
@@ -0,0 +1,50 @@
+/// \file
+/// \brief This file implements CLI initialized configuration.
+
+#include "nixd/CommandLine/Configuration.h"
+#include "nixd/CommandLine/Options.h"
+#include "nixd/Controller/Configuration.h"
+
+#include "lspserver/Logger.h"
+
+#include <llvm/Support/CommandLine.h>
+#include <llvm/Support/JSON.h>
+
+#include <iostream>
+#include <string>
+
+using namespace nixd;
+using namespace llvm::cl;
+
+namespace {
+
+opt<std::string> DefaultConfigJSON{"config",
+                                   desc("JSON-encoded initial configuration"),
+                                   init(""), cat(NixdCategory)};
+
+} // namespace
+
+Configuration nixd::parseCLIConfig() {
+  if (DefaultConfigJSON.empty())
+    return {};
+
+  llvm::Expected<llvm::json::Value> V = llvm::json::parse(DefaultConfigJSON);
+
+  if (!V) {
+    llvm::errs()
```

Comment:
This error seems like something the caller should handle. Reporting the error here would make this function less versatile.